### PR TITLE
feat(deploy-service): Added check cluster configuration warning items

### PR DESCRIPTION
**Pull request by Luckyboys**

### CODE REVIEW

File: pkg/service/api/v1/deploy/checking.go [hunk 1]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+	wizardData := wizard.GetCurrentWizard()
+	if wizardData.Info.NodePortMinimum <= suggestNodePortMinimum {
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: fmt.Sprintf("Node port minimum setting too small warning"), // NodePort 最小端口号设置过小警告
+			Result:        constant.CheckResultWarning,
+			Error: &api.Error{
+				Reason:     "Node port minimum setting too small",                                                             // NodePort 最小端口号太小
+				Detail:     fmt.Sprintf("We suggest node port minimum port should be higher than %d", suggestNodePortMinimum), // 我们建议 NodePort 最小端口号大于 %d
+				FixMethods: fmt.Sprintf("Modify the node port minimum higher than %d", suggestNodePortMinimum),                // 修改最小端口号为大于 %d
```
Comment (yangruiray) [inline pkg/service/api/v1/deploy/checking.go]:
```go
fmt.Sprintf("Modify the node port larger than %d", suggestNodePortMinimun)
```

File: pkg/service/api/v1/deploy/checking.go [hunk 2]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+}
+
+func checkClusterNodePortInterval() (warnings []*api.CheckingItem) {
+
+	warnings = make([]*api.CheckingItem, 0)
+	wizardData := wizard.GetCurrentWizard()
+	if wizardData.Info.NodePortMaximum-wizardData.Info.NodePortMinimum > suggestNodePortMaxInterval {
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: "Node port interval setting too large warning", // NodePort 监控区间设置过大警告
```
Comment (yangruiray) [inline pkg/service/api/v1/deploy/checking.go]: 不用后面的 warning，感觉读不通

File: pkg/service/api/v1/deploy/checking.go [hunk 3]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+			Result:        constant.CheckResultWarning,
+			Error: &api.Error{
+				Reason:     "Node port interval setting too large",                                                           // NodePort 范围过大
+				Detail:     fmt.Sprintf("We suggest node port interval should be small than %d", suggestNodePortMaxInterval), // 我们建议 NodePort 端口号范围应该小于 %d
```
Comment (yangruiray) [inline pkg/service/api/v1/deploy/checking.go]: small => smaller

File: pkg/service/api/v1/deploy/checking.go (hunk 2 above)
Comment (zyseeker) [inline pkg/service/api/v1/deploy/checking.go]: "Node port interval setting too large warning" -> "Warning of too large value on node port interval setting"

File: pkg/service/api/v1/deploy/deploy.go [hunk 4]
```diff
@@ -50,6 +50,11 @@ func Deploy(c *gin.Context) {
 		return
 	}
 
+	if !checkClusterConfiguration() {
+		h.E(c, h.EStatusError.WithPayload("current cluster configuration check status is not passed"))
```
Comment (zyseeker) [inline pkg/service/api/v1/deploy/deploy.go]: "status"可以去掉，有点多余

File: pkg/service/api/v1/deploy/checking.go [hunk 5]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+
+	warnings = make([]*api.CheckingItem, 0)
+	wizardData := wizard.GetCurrentWizard()
+	if wizardData.Info.NodePortMinimum <= suggestNodePortMinimum {
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: fmt.Sprintf("Node port minimum setting too small warning"), // NodePort 最小端口号设置过小警告
+			Result:        constant.CheckResultWarning,
+			Error: &api.Error{
+				Reason:     "Node port minimum setting too small",                                                             // NodePort 最小端口号太小
```
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go [hunk 6]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+	}
+
+	return
+}
+
+func checkClusterNodePortMinimum() (warnings []*api.CheckingItem) {
+
+	warnings = make([]*api.CheckingItem, 0)
+	wizardData := wizard.GetCurrentWizard()
+	if wizardData.Info.NodePortMinimum <= suggestNodePortMinimum {
```
[the remark this answers is not in the document]
Comment (randmonkey) [reply pkg/service/api/v1/deploy/checking.go]: 等于应该可以

[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/deploy.go (hunk 4 above)
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/deploy.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 2 above)
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go [hunk 7]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+		}
+	}
+	for role, counter := range counters {
+
+		if counter >= haCount[role] {
+			continue
+		}
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: fmt.Sprintf("Nodes for %s high available", role),
```
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go [hunk 8]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+}
+
+func checkClusterNodePortMinimum() (warnings []*api.CheckingItem) {
+
+	warnings = make([]*api.CheckingItem, 0)
+	wizardData := wizard.GetCurrentWizard()
+	if wizardData.Info.NodePortMinimum <= suggestNodePortMinimum {
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: fmt.Sprintf("Node port minimum setting too small warning"), // NodePort 最小端口号设置过小警告
```
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 3 above)
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 2 above)
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 1 above)
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go [hunk 9]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+			Result:        constant.CheckResultWarning,
+			Error: &api.Error{
+				Reason:     "Node port minimum setting too small",                                                             // NodePort 最小端口号太小
+				Detail:     fmt.Sprintf("We suggest node port minimum port should be higher than %d", suggestNodePortMinimum), // 我们建议 NodePort 最小端口号大于 %d
```
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

File: pkg/service/api/v1/deploy/checking.go [hunk 10]
```diff
@@ -270,8 +273,98 @@ func checkWrongClusterConfiguration() (errs []*api.CheckingItem) {
...
+		if counter >= haCount[role] {
+			continue
+		}
+
+		warnings = append(warnings, &api.CheckingItem{
+			CheckingPoint: fmt.Sprintf("Nodes for %s high available", role),
+			Result:        constant.CheckResultWarning,
+			Error: &api.Error{
+				Reason:     fmt.Sprintf("nodes for %s are not enough for high available", role),                                     // %s 角色的节点数不满足高可用数量
+				Detail:     fmt.Sprintf("We suggest nodes for %s count should be equal or more than %d nodes", role, haCount[role]), // 我们建议 %s 角色节点数大于等于%d 个
```
[the remark this answers is not in the document]
Comment (Luckyboys) [reply pkg/service/api/v1/deploy/checking.go]: Fixed

**Pull request merged**